README.md

```diff
@@ -398,6 +399,49 @@ const ui = (
 
 This callback will only be called if `isOpen` is `true`.
 
+### modifyStateChange
+
+> `function(state: object, stateToBeSet: object)` | optional
+
+**🚨 This is a really handy power feature 🚨**
+
+This function will be called each time `downshift` sets its internal state
+(or calls your `onStateChange` handler for control props). It allows you to
+modify the state change that will take place which can give you fine grain
+control over how the component interacts with user updates without having to
+use [Control Props](#control-props).
+
+* `state`: The full current state of downshift.
+* `stateToBeSet`: The state that is about to be set (including the `type` or
+  origin of the change which has references in `Downshift.stateChangeTypes`
+  as in [`onStateChange`](#onstatechange).
+
+This is a pure function. You _might_ think of it as a simple reducer!
```

Comment:
This is where I decided to switch the function signature to accept state first. I think some folks will get giddy about this :wink:

Reply:
Right. State first its better; like simple reducer.